src/app/logger.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright 2018 Novo Nordisk Foundation Center for Biosustainability, DTU.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {environment} from '../environments/environment';
+
+class IGAEvent {
+  public category?: string;
+  public action?: string;
+  public label?: string;
+  public value?: string;
+}
+
+export class GAEvent extends IGAEvent {
+  constructor(event: IGAEvent) {
+    super();
+    Object.assign(
+      this,
+      {
+        category: null,
+        action: null,
+        label: null,
+        value: null,
+      },
+      event,
+    );
+  }
+}
+
+// tslint:disable-next-line:no-any
+export const debug = (...args: any[]): void => {
+  if (!environment.production) {
+    console.log('Debug:', ...args);
+  }
+};
+
+/**
+ * It is often required to prefix the log messages somehow.
+ * It's useful for filtering them in the developer console.
+ * This method provides a convenient way to create a curried logger.
+ * @param tag the prefix for the log messages
+ */
+export const debugC = (tag: string): (...args: string[]) => void =>
+    (...args: string[]): void => {
+      debug(tag, ...args);
+    };
+
+export function info(gaEvent: GAEvent): void;
```

Comment:
I'd perhaps consider a separate `event` function, unrelated to the log levels. Then it can accept only `GAEvent` and you don't have to check instance type etc. Since GA events don't have a concept of loglevels I think it makes sense not to mix events and loglevels.